### PR TITLE
Push Resource Usage stats to remote sinks

### PR DESCRIPTION
**Pull request by diptanu**

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -1338,9 +1340,37 @@ func (c *Client) collectHostStats() {
 			c.resourceUsageLock.RLock()
 			c.resourceUsage.Enqueue(ru)
 			c.resourceUsageLock.RUnlock()
+			if ru != nil {
```
Comment (dadgar) [inline client/client.go]: You probably shouldn't enqueue the nil `ru`.

File: client/task_runner.go [hunk 2]
```diff
@@ -475,6 +477,9 @@ func (r *TaskRunner) collectResourceUsageStats() {
 			r.resourceUsageLock.Lock()
 			r.resourceUsage.Enqueue(ru)
 			r.resourceUsageLock.Unlock()
+			if ru != nil {
```
Comment (dadgar) [inline client/task_runner.go]: Same here

**Comment by dadgar** [comment]
LGTM

**Comment by dadgar** [comment]
Can we add docs to telemetry section that the client does this

**Comment by diptanu** [comment]
@dadgar Following up with another PR

**Pull request merged**